### PR TITLE
First pass at a guide for setting up Travis-CI

### PR DESCRIPTION
**Pull request by briehl**
See also here: https://atlassian.kbase.us/browse/KBASE-1949

This is a first pass at setting up Travis-CI and Coveralls. The Coveralls section is empty right now. I'll start to fill those sections in tomorrow, but wanted to get this started. There will obviously be challenged in getting KBase code to run in a reasonably bare and stateless environment like Travis-CI. Mike and Roman outlined some issues they ran into while setting up the Method Store repo, and I added a couple thoughts during my own exploration.

Anyway, open for discussion!

### CODE REVIEW

File: Setting_up_Travis_CI_and_Coveralls.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+branches:
+  only:
+  - master
+  - staging
+  - develop
+  - dev
+```
+
+The test script is written in Perl, so it sets the language as Perl. The ```install:``` command runs each proceeding line in order before running anything. Note that this can include a ```git clone``` line to pull in any dependencies.
```
Comment (fperez) [inline Setting_up_Travis_CI_and_Coveralls.md]: Minor comment: when putting code blocks inline, a single pair of backticks should suffice, triple backticks are only needed for multiline blocks (as you used above). It keeps the document a bit more readable in source form.

File: Setting_up_Travis_CI_and_Coveralls.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+[Background and automated testing](#background)  
+[Travis-CI initial setup](#travis-init)  
+[Travis-CI script configuration](#travis-config)  
+[Coveralls initial setup](#coveralls-init)  
+[Coveralls configuration](#coveralls-config)  
+[KBase Challenges](#kbase-challenges)  
+
+## <a name="background"></a>Background and automated testing
+
+Using Github gives us numerous tools to help with automatic unit testing of open source projects. Travis-CI (https://travis-ci.org) automatically downloads and tests your repo, returning whether or not all tests pass. If you tests output a .coverage file (see [below](#coveralls-config)), that can be automatically passed to Coveralls (https://coveralls.io/) where the amount of code tested can be reported in your repo.
```
Comment (fperez) [inline Setting_up_Travis_CI_and_Coveralls.md]: I'd make the links here regular MD urls like `[Travis-CI](https://travis-ci.org)`, that way the rendered doc feels even more 'html-like' :)

**Comment by fperez** [comment]
Same general comment about using regular `[...](..)` syntax for most URLs, it will make the overall doc read better when rendered.

Very minor edits, otherwise awesome! Many thanks, we can merge this once these minor edits are made.

**Comment by briehl** [comment]
Thanks for the feedback. I'll make some edits.

**Comment by briehl** [comment]
Okay, I added those changes. I also put in a header for looking up test results, since that wasn't terribly clear to me at first.

**Comment by dangunter** [comment]
+1 to merge. I think we should merge this now, so others can easily add to it

**Comment by fperez** [comment]
Sorry for the delay!!! Merging now.

**Pull request merged**